### ISSUE TITLE
Treeselect wont let you remove a child after removing a sibling

### ISSUE THREAD

**Issue by matgargano** [ant-design/ant-design#17686]
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/distracted-fire-vkxdq)

### Steps to reproduce
on https://codesandbox.io/s/distracted-fire-vkxdq

put mouse in treeselect
select “STEP 1 CLICK ME”
click x next to *STEP 2 X ME OFF *
click x next to *STEP 3 (final step) X ME OFF notice how i don’t go away!*

Also notice how 994490039 exists twice in the selected part of the tree select as do a bunch of others (994490040, 994490044, 994490044 etc) 

### What is expected?
when I click on *STEP 3 (final step) X ME OFF notice how i don’t go away!* it should go away and be removed from the state

### What is actually happening?
*STEP 3 (final step) X ME OFF notice how i don’t go away!* remains

| Environment | Info |
|---|---|
| antd | 3.20.3 |
| React | latest |
| System | MacOS |
| Browser | Chrome Version 75.0.3770.100 (Official Build) (64-bit) |

**Comment by zombieJ** [comment]
Duplicate of #17615

**Comment by matgargano** [comment]
@zombieJ  can you explain what the fix is? #17615 is closed and references https://github.com/react-component/animate/commit/9a9f7946245db629713b45d194eb43ccdc636649 which is closed -- but using 

```
{
  "dependencies": {
    "react-dom": "latest",
    "react": "latest",
    "antd": "3.20.3"
  }
}

```

I still see the bug persist...  see https://codesandbox.io/s/quiet-forest-gd763

**Comment by zombieJ** [comment]
@matgargano  reinstall `node_modules`: https://codesandbox.io/s/dank-moon-dcj1w
